File: src/main/java/me/zeroeightsix/kami/module/modules/movement/AntiLevitation.java
package me.zeroeightsix.kami.module.modules.movement;

import me.zeroeightsix.kami.module.Module;
import net.minecraft.potion.Potion;

/**
 * Created by 0x2E | PretendingToCode
 */
@Module.Info(
        name = "AntiLevitation",
        description = "Removes levitation potion effect",
        category = Module.Category.MOVEMENT
)
public class AntiLevitation extends Module {

    @Override
    public void onUpdate() {
        if (mc.player.isPotionActive(Potion.getPotionFromResourceLocation("levitation"))) {
            mc.player.removeActivePotionEffect(Potion.getPotionFromResourceLocation("levitation"));
        }
    }
}

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/AntiLevitation.kt
package me.zeroeightsix.kami.module.modules.movement

import me.zeroeightsix.kami.module.Module
import me.zeroeightsix.kami.util.threads.safeListener
import net.minecraft.init.MobEffects
import net.minecraftforge.fml.common.gameevent.TickEvent

object AntiLevitation : Module(
    name = "AntiLevitation",
    description = "Removes levitation potion effect",
    category = Category.MOVEMENT
) {
    init {
        safeListener<TickEvent.ClientTickEvent> {
            if (player.isPotionActive(MobEffects.LEVITATION)) {
                player.removeActivePotionEffect(MobEffects.LEVITATION)
            }
        }
    }
}